services/tables/src/main/java/com/linkedin/openhouse/tables/controller/TablesController.java

```diff
@@ -97,6 +97,38 @@ public ResponseEntity<GetAllTablesResponseBody> searchTables(
         apiResponse.getResponseBody(), apiResponse.getHttpHeaders(), apiResponse.getHttpStatus());
   }
 
+  @Operation(
+      summary = "Search Tables in a Database",
+      description =
+          "Returns a list of Table resources present in a database. Only filter supported is 'database_id'.",
+      tags = {"Table"})
+  @ApiResponses(
+      value = {
+        @ApiResponse(responseCode = "200", description = "Table SEARCH: OK"),
+        @ApiResponse(responseCode = "400", description = "Table SEARCH: BAD_REQUEST"),
+        @ApiResponse(responseCode = "401", description = "Table SEARCH: UNAUTHORIZED"),
+        @ApiResponse(responseCode = "403", description = "Table SEARCH: FORBIDDEN"),
+        @ApiResponse(responseCode = "404", description = "Table SEARCH: NOT_FOUND")
+      })
+  @PostMapping(
+      value = {
+        "/v0/databases/{databaseId}/tables/search",
+        "/v1/databases/{databaseId}/tables/search"
+      },
+      produces = {"application/json"})
+  public ResponseEntity<GetAllTablesResponseBody> searchTablesPaginated(
+      @Parameter(description = "Database ID", required = true) @PathVariable String databaseId,
+      @Parameter(description = "Page index", required = true) @PathVariable int page,
+      @Parameter(description = "Page size", required = true) @PathVariable int size,
+      @Parameter(description = "SortBy field") @PathVariable String sortBy) {
```

Comment:
SortBy field can be non mandatory. Usually it is `databaseId` or `tableId` unless specified field is mentioned. So we can default to `databaseId`. So this field can be nullable as it is inferred in the code if not specified.